remote.go

```diff
@@ -454,6 +454,25 @@ func (o *Remote) PushUrl() string {
 	return C.GoString(C.git_remote_pushurl(o.ptr))
 }
 
+func (c *RemoteCollection) Rename(remote, newname string) ([]string, error) {
+	cproblems := C.git_strarray{}
+	defer freeStrarray(&cproblems)
+	cnewname := C.CString(newname)
+	defer C.free(unsafe.Pointer(cnewname))
+	cremote := C.CString(remote)
+	defer C.free(unsafe.Pointer(cremote))
+
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	ret := C.git_remote_rename(&cproblems, c.repo.ptr, cremote, cnewname)
+	if ret < 0 {
+		problems := makeStringsFromCStrings(cproblems.strings, int(cproblems.count))
+		return problems, MakeGitError(ret)
```

Comment:
An error indicates an error in the function, but failure to rename some refspecs because they do not follow the standard pattern is not an error condition. The error array is returned when `ret == 0`. As the documentation for this function says, the ownership of the strings passes to the caller, so you need to free the string array.